README.md

```diff
@@ -33,69 +47,225 @@
       stop_name text, -- The name of the station
       estimated_pop_800m integer, -- The population within 800 meters
       the_geom geometry(Point, 4326) -- The geometry of the bus stop
+     create index septa_bus_stops__the_geom__32129__idx
+    on septa_bus_stops
+    using GiST (ST_Transform(the_geom, 32129));
+UPDATE septa_bus_stops
+    set the_geom = st_setsrid(st_makepoint(stop_lon, stop_lat), 4326)
+with septa_bus_stop_block_groups as (
+    select
+        s.stop_id,
+        '1500000US' || bg.geoid10 as geo_id
+    from septa_bus_stops as s
+    join census_block_groups as bg
+        on ST_DWithin(
+            ST_Transform(s.the_geom, 32129),
+            ST_Transform(bg.the_geom, 32129),
+            800
+        )
+),
+septa_bus_stop_surrounding_population as (
+    select
+        stop_id,
+        sum(total) as estimated_pop_800m
+    from septa_bus_stop_block_groups as s
+    join census_population as p on s.geo_id = p.id
+    group by stop_id
+)
+select
+    stop_name,
+    estimated_pop_800m,
+    the_geom
+from septa_bus_stop_surrounding_population
+join septa_bus_stops using (stop_id)
+order by estimated_pop_800m asc
+limit 1
   )
   ```
 
-3. Using the Philadelphia Water Department Stormwater Billing Parcels dataset, pair each parcel with its closest bus stop. The final result should give the parcel address, bus stop name, and distance apart in meters. Order by distance (largest on top).
-
+@@ -41,9 +115,39 @@
   **Structure:**
   ```sql
   (
       address text,  -- The address of the parcel
       stop_name text,  -- The name of the bus stop
       distance_m double precision  -- The distance apart in meters
+      -- create a geometry column on buses
+alter table septa_bus_stops
+    add column if not exists geometry geometry(Point, 4326);
+update septa_bus_stops
+  set geometry = ST_SetSRID(ST_MakePoint(stop_lon, stop_lat), 4326);
+  
+-- indexes
+create index if not exists septa_bus_stops__the_geom__32129__idx
+    on septa_bus_stops
+    using GiST (ST_Transform(geometry, 32129));
+create index if not exists phl_pwd_parcels_geometry_32129_idx
+    on phl_pwd_parcels
+    using GiST (ST_Transform(geometry, 32129));
+-- query & cross join
+SELECT 
+    p.address,
+    cl.stop_name,
+    cl.distance_m
+from phl_pwd_parcels as p
+cross join lateral(
+    select 
+        s.stop_name,
+        ST_Distance(
```

Comment:
Note that `ST_Distance` isn't able to take advantage of spatial indexing; for that you would use the nearest neighbor operator (`<->`)